nautobot/utilities/views.py

```diff
@@ -151,3 +187,1051 @@ def get_return_url(self, request, obj=None):
+
+                    # Check that the new object conforms with any assigned object-level permissions
+                    self.queryset.get(pk=obj.pk)
+
+                msg = f'{"Created" if object_created else "Modified"} {self.queryset.model._meta.verbose_name}'
+                logger.info(f"{msg} {obj} (PK: {obj.pk})")
+                if hasattr(obj, "get_absolute_url"):
+                    msg = f'{msg} <a href="{obj.get_absolute_url()}">{escape(obj)}</a>'
+                else:
+                    msg = f"{msg} { escape(obj)}"
+                messages.success(request, mark_safe(msg))
+
+                if "_addanother" in request.POST:
+
+                    # If the object has clone_fields, pre-populate a new instance of the form
+                    if hasattr(obj, "clone_fields"):
+                        url = f"{request.path}?{prepare_cloned_fields(obj)}"
+                        return redirect(url)
+
+                    return redirect(request.get_full_path())
+
+                return_url = form.cleaned_data.get("return_url")
+                if return_url is not None and is_safe_url(url=return_url, allowed_hosts=request.get_host()):
+                    return redirect(return_url)
+                else:
+                    return redirect(self.get_return_url(request, obj))
+
+            except ObjectDoesNotExist:
+                msg = "Object save failed due to object-level permissions violation"
+                logger.debug(msg)
+                form.add_error(None, msg)
+
+        else:
+            logger.debug("Form validation failed")
+
+        return render(
+            request,
+            self.template_name,
+            {
+                "obj": obj,
+                "obj_type": self.queryset.model._meta.verbose_name,
+                "form": form,
+                "return_url": self.get_return_url(request, obj),
+                "editing": obj.present_in_database,
+            },
+        )
+
+
+class ObjectDeleteViewMixin(GetReturnURLMixin, NautobotViewSetMixin):
+    """
+    Delete a single object.
+    queryset: The base queryset for the object being deleted
+    template_name: The name of the template
+    """
+
+    def define_routes(self):
+        return super().define_routes() + [
+            Route(
+                url=r"^{prefix}/{lookup}/delete/$",
+                mapping={
+                    "get": "handle_object_delete_get",
+                    "post": "handle_object_delete_post",
+                },
+                name="{basename}_delete",
+                detail=True,
+                initkwargs={"suffix": "Delete"},
+            ),
+        ]
+
+    def get_object_for_delete(self, kwargs):
+        # Look up object by slug if one has been provided. Otherwise, use PK.
+        if "slug" in kwargs:
+            return get_object_or_404(self.queryset, slug=kwargs["slug"])
+        else:
+            return get_object_or_404(self.queryset, pk=kwargs["pk"])
```

Comment:
Again, I am not clear on need for the duplication here? Can't all the HTTP verbs/actions just rely on `get_object()` seeing as you're never doing more than one per request?